service.py

```diff
@@ -300,7 +300,9 @@ def check_mail(mail):
 def get_subject(subjectID):
     subject = Subject.query.filter_by(id=subjectID).first()
     url = Tool.subject_url(subject)
-    articles = Article.query.filter_by(subject_id=subject.id, status=1).all()
+    page = request.args.get('page', 1, type=int)        #Set the default page to page 1
+    articles = Article.query.filter_by(subject_id=subject.id, status=1).order_by(Article.time.desc()).paginate(page=page, per_page=5)
```

Comment:
what does `paginate(page=page, per_page=5)` do ? 

Thanks.

Reply:
@hema-001 
Here we don't want to retrieve all the articles, we just want a certain number of articles from a given page. Therefore, `paginate(page=page, per_page=5)` retrieves 5 articles from the page `page`. The variable `page` is the page we want to retrieve our items from and `per_page` is the number of articles on each page. The line I removed was using `all()` which means it returns a `list` object, but `paginate()` returns a `flask_sqlalchemy.Pagination` object.

Reply:
So basically, `paginate()` returns a page that contains a list of articles.